reference/api/keys.md

```diff
@@ -1,30 +1,277 @@
 
-You must have the master key to access the `keys` route.
+```bash
+curl \
+  -H 'Authorization: Bearer replaceThisWithYourMasterKey'
+  …
+```
 
-[More information about the keys and their rights](/reference/features/authentication.md).
+[Learn more about managing keys and their rights](/reference/features/authentication.md).
 
-## Get keys
+## Get all keys
 
 <RouteHighlighter method="GET" route="/keys"/>
 
-Get the **private** and **public** [key](/reference/features/authentication.md).
+List all existing API keys. **Expired keys are included in the response**, but deleted keys are not.
+
+[See below for an explanation of returned fields.](#description)
+
+### Example
+
+<CodeSamples id="get_all_keys_1" />
+
+#### Response: `200 Ok`
+
+```json
+{
+  "results": [
+    {
+      "description": "Manage documents: Products/Reviews API key",
+      "key": "d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4",
+      "actions": [
+        "documents.add",
+        "documents.delete"
+      ],
+      "indexes": [
+        "products",
+        "reviews"
+      ],
+      "expiresAt": "2021-12-31T23:59:59Z",
+      "createdAt": "2021-10-12T00:00:00Z",
+      "updatedAt": "2021-10-13T15:00:00Z"
+    },
+    {
+      "description": "Default Search API Key (Use it to search from the frontend code)",
+      "key": "0a6e572506c52ab0bd6195921575d23092b7f0c284ab4ac86d12346c33057f99",
+      "actions": [
+        "search"
+      ],
+      "indexes": [
+        "*"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    },
+    {
+      "description": "Default Admin API Key (Use it for all other operations. Caution! Do not share it on the client side)",
+      "key": "380689dd379232519a54d15935750cc7625620a2ea2fc06907cb40ba5b421b6f",
+      "actions": [
+        "*"
+      ],
+      "indexes": [
+        "*"
+      ],
+      "expiresAt": null,
+      "createdAt": "2021-08-11T10:00:00Z",
+      "updatedAt": "2021-08-11T10:00:00Z"
+    }
+  ]
+}
+```
+
+#### `description`
+
+A description for the key. Default value is `null`.
+
+#### `key`
+
+An alphanumeric key value generated by MeiliSearch on API key creation. Used for authorization when [making calls to a protected MeiliSearch instance](/reference/features/authentication.md#communicating-with-a-protected-instance).
+
+This value is also used as the `:key` path variable to [update](#update-a-key), [delete](#delete-a-key), or [get](#get-one-key) a specific key.
+
+#### `actions`
+
+An array of API actions permitted for the key. `["*"]` for all actions.
+
+The `*` character can also be used as a wildcard to get all permissions for a route, e.g. `documents.*` to authorize access to all document-related endpoints.
+
+#### `indexes`
+
+An array of indexes permitted for the key. `["*"]` for all indexes.
+
+#### `expiresAt`
+
+Date and time when the key will expire, represented in **ISO 8601** format. `null` if the key never expires.
```

Comment:
Do we want ISO 8601 to be bold every time?

Reply:
Good catch. I'm against it, feels like we're giving it too much importance.